Code/Source/solver/mat_models.cpp

```diff
@@ -188,116 +360,120 @@ void get_pk2cc(const ComMod& com_mod, const CepMod& cep_mod, const dmnType& lDmn
   }
 
   // Electromechanics coupling - active strain
-  auto Fe  = F;
-  auto Fa = mat_id(nsd);
-  auto Fai = Fa;
-
-  if (cep_mod.cem.aStrain) {
-    actv_strain(com_mod, cep_mod, ya, nfd, fl, Fa);
-    Fai = mat_inv(Fa, nsd);
-    Fe = mat_mul(F, Fai);
-  }
-
-  double J = mat_det(Fe, nsd);
+  EigenMatrix<nsd> Fe  = F;
+  EigenMatrix<nsd> Fa = EigenMatrix<nsd>::Identity();
+  EigenMatrix<nsd> Fai = Fa;
+
+  // if (cep_mod.cem.aStrain) {
+  //   actv_strain(com_mod, cep_mod, ya, nfd, fl, Fa);
+  //   Fai = Fa.inverse();
+  //   Fe = F * Fai;
+  // }
+
+  Ja = Fa.determinant();
+  double J = Fe.determinant();
   double J2d = pow(J, (-2.0/nd));
   double J4d = J2d*J2d;
 
-  auto Idm = mat_id(nsd);
-  auto C = mat_mul(transpose(Fe), Fe);
-  Array<double> E(nsd,nsd);
-  for (int i = 0; i < nsd; i++) {
-    for (int j = 0; j < nsd; j++) {
-      E(i,j) = 0.50 * (C(i,j) - Idm(i,j));
+  EigenMatrix<nsd> Idm = EigenMatrix<nsd>::Identity();
+  EigenMatrix<nsd> C = Fe.transpose() * Fe;
+  EigenMatrix<nsd> E = 0.50 * (C - Idm);
+
```

Comment:
It's not clear to me what an `EigenMatrix<nsd>` is. Is this a vector or an nsd x nsd array ?

Reply:
`EigenMatrix<nsd>` is an nsd x nsd array. `EigenMatrix` is an alias defined at the top of the file.

template<size_t nsd>
using EigenMatrix = Eigen::Matrix<double, nsd, nsd>;

Open to any suggestions for a different alias name!

Reply:
@aabrown100-git My mistake, confused `EigenMatrix<nsd>` with an actual Eigen object, looked like a vector to me.

I recommend using `Matrix` rather than `EigenMatrix`, same for `Tensor` and `Vector`.

Reply:
Sounds good, I changed EigenMatrix to Matrix, and EigenTensor to Tensor. I actually don't use an Eigen vector so didn't need to define a Vector alias.